packages/types/src/types/provider/vsphere/VM.ts

```diff
@@ -20,4 +20,6 @@ export interface VSphereVM extends VSphereResource {
   disks: VSphereDisk[];
   // Concerns          []model.Concern `json:"concerns"`
   concerns: Concern[];
+  // prop added by the UI to implement narrowing (discriminated union)
+  providerType: 'vsphere';
```

Comment:
what are the pros and cons of adding the  `providerType`  in the vm interfaces ?
what are the pros and cons of adding the  `providerType`  in the `VSphereResource`  (e.g. the resource interface) interfaces ?

I prefer to add it fo the resource interface, if we are using the resource interface (parent of all resources) we get the `kind` field for all resources, if we add it to the vm we will need to add it to other resources as we go foreword ?

Reply:
Moved the prop to the Resource interface (where possible).

@yaacov 
As for the pros&cons: similar as with adding the prop to the  VM interface:
1. encouraging different programming style -  per-provider switch
2. problems with populating the field - backend is not providing the field so it needs to be added after fetching the data. For VMs it's delegated to the hook. In other cases we need a solution (right now the field is optional)

There is one extra problem: all providers inherit from `OpenshiftResource` so OpenShift related entities need a different solution. As for now the prop was added only to the OpenShift VM.

Reply:
can you make a `baseResource` and make `OpenshiftResource` and all the others base from ?